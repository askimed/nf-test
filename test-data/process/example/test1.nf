process test {

  publishDir "${params.outdir}", mode: 'copy'

  input:
    val number

  output:
     file "*.txt"

  script:
  if (params.outdir == null) {
    exit 1 "No outdir set"
  }


  """
  echo "lukas forer ${number}" > "${number}_${params.lukas}.txt"
  """

}
