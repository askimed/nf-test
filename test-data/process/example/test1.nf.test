nextflow_process {

  name "Test test1.nf"

  script "./test1.nf"
  process "test"

  test("Should create 5 output files") {

   when {
     params.lukas = 28
     params.forer = "22"
     params.outdir = "${outputDir}"
     process {
       """
       input[0]=Channel.of(1..5)
       """
     }  
   }

   then {
     assert workflow.success
     assert params.lukas == 28
     for (def i in 1..5){
       def file = new File("${params.outdir}/${i}_28.txt");
       assert file.exists()
       assert file.text == "lukas forer ${i}\n"
     }
     assert snapshot(
        workflow.success, path("${params.outdir}").list()
     ).match("process_test_1")
   }

  }


  test("Should create 5 output files from params file") {

   when {
     loadParams("$moduleDir/params.json")
     process {
       """
       input[0]=Channel.of(1..5)
       """
     }     
   }

   then {
     assert workflow.success
     assert params.lukas == 28
     for (def i in 1..5){
       def file = new File("${params.outdir}/${i}_28.txt");
       assert file.exists()
       assert file.text == "lukas forer ${i}\n"
     }
   }

  }
  
  test("Should create 5 output files from params file 2") {

   when {
     params {
       load("$moduleDir/params.json")
       lukas = 44
     }
     process {
       """
       input[0]=Channel.of(1..5)
       """
     }     
   }

   then {
     assert workflow.success
     assert params.lukas == 44
     for (def i in 1..5){
       def file = new File("${params.outdir}/${i}_44.txt");
       assert file.exists()
       assert file.text == "lukas forer ${i}\n"
     }
   }

  }

  test("Should create 5 output files from yaml file") {

   when {
     params {
       load("$moduleDir/params.yml")
     }
     process {
       """
       input[0]=Channel.of(1..5)
       """
     }       
   }

   then {
     assert workflow.success
     assert params.lukas == 28
     for (def i in 1..5){
       def file = new File("${params.outdir}/${i}_28.txt");
       assert file.exists()
       assert file.text == "lukas forer ${i}\n"
     }
   }

  }

  test("Should fail if outdir is missing") {

    //debug true

    when {
      params.lukas = 27
      params.forer = "22"
      process {
        """
        input[0]=Channel.of(1..5)
        """
      }     
    }

    then {
      assert workflow.failed
      assert params.lukas == 27
      assert params.forer == "22"
    }

  }

}
