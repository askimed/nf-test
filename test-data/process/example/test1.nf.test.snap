{
    "process_test_1": {
        "content": [
            true,
            [
                "1_28.txt:md5,cb6d376c512ae5d94a5b4031a6d84441",
                "2_28.txt:md5,14a244f8e6f13b3429711200c7e89bf4",
                "3_28.txt:md5,b2f4edba5545c31d5b8e246136a12c63",
                "4_28.txt:md5,74136f1ebff66a01335c4f7bb0e11582",
                "5_28.txt:md5,b6713b59c66f67fd6a28e8b09ff4f582"
            ]
        ],
        "timestamp": "2023-09-02T15:11:26.550772"
    }
}